packages/kbn-build/src/utils/packages.js

```diff
@@ -0,0 +1,137 @@
+import _glob from 'glob';
+import path from 'path';
+import { promisify } from 'bluebird';
+
+import { CliError } from './errors';
+import { Package } from './Package';
+
+const glob = promisify(_glob);
+
+export async function getPackages(rootPath, packagesPaths) {
+  const globOpts = {
+    cwd: rootPath,
+    strict: true,
+    absolute: true
+  };
+  const packages = new Map();
+
+  for (const globPath of packagesPaths) {
+    const res = await glob(path.join(globPath, 'package.json'), globOpts);
+
+    for (const globResult of res) {
+      const packageConfigPath = normalize(globResult);
+      const packageDir = path.dirname(packageConfigPath);
+      const pkg = await Package.fromPath(packageDir);
+
+      packages.set(pkg.name, pkg);
```

Comment:
__question:__ wondering if we should emit warning if `packages` already contains package with the same name, theoretically it's possible, but not sure if it makes sense to handle this case specifically?

Reply:
I made it throw an error. I think it can happen if someone copy-pastes another plugin to get started with a one.